### ISSUE TITLE
Move to mariadb 3.x driver

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#29916]
There seems to be still a mysql connection string compatibility if we include https://mariadb.com/kb/en/about-mariadb-connector-j/#jdbcmysql-scheme-compatibility

**Comment by perivamsi** [comment]
@camsaul warns that this might not be a good idea because of some issues with the latest mariadb driver

**Comment by camsaul** [comment]
Yeah, the 3.x driver made a change where if you `INSERT` multiple rows `RETURNING` the created primary keys, it will only return the PK for the first row (unlike 2.x which returns the PKs for all the rows). This will probably break some of our Toucan stuff and Actions stuff

**Comment by paoliniluis** [comment]
There’s something we can include in the connection string to make the driver compatible:
returnMultiValuesGeneratedIds

https://mariadb.com/kb/en/about-mariadb-connector-j/#infrequently-used-parameters

**Comment by e-gaulue** [comment]
I suppose you may know : EOL for ConnectorJ 2.7 is September 2025. (https://mariadb.com/kb/en/eol-for-mariadb-connectorj-27-mariadb-java-client-27/).

I don't know how long it takes to integrate version 3 to Metabase and test it enough to be sure there is no regression. If version 3 behave differently than version 2, all blocking points should be identified and fixed, because anyway we are not going to be stuck on 2.7 forever. The later we will try/move, the later we may discover some.

Could we imagine a branch or an option to compile for version 3 and help to identify those things to make better.

**Comment by camsaul** [comment]
Oh ok `returnMultiValuesGeneratedIds` must be new I don't think it existed a year or two ago, last time I looked at switching to 3.x
